### PR TITLE
Improve use of curses windows and log messages

### PR DESCRIPTION
**Pull request by Bidski**
Creates two curses windows on the screen.

 - A 5 line window at the top of the screen to display current walk command and status.
 - A second window occupying the rest of the screen to display log messages.

Log messages should hopefully display with the appropriate colour as they normally would.

The log messages window should scroll off the bottom correctly, but any lines that scroll off the top will be lost. FileLogHandler should be coupled with these changes if log messages want to be kept.

image

### CODE REVIEW

**Comment by JosephusPaye** [comment]
This is nice! How would you enable file logging with this if you want to keep log messages?

**Comment by Bidski** [comment]
FileLogHandler module, but I have had issues using it with docker, needs some more playing with

File: module/behaviour/strategy/KeyboardWalk/src/KeyboardWalk.cpp [hunk 1]
```diff
@@ -77,64 +120,166 @@ namespace behaviour {
...
+
+            on<Trigger<LogMessage>, Sync<KeyboardWalk>>().then([this](const LogMessage& packet) {
+                // Where this message came from
+                std::string source = "";
+
+                // If we know where this log message came from, we display that
+                if (packet.task) {
+                    // Get our reactor name
+                    std::string reactor = packet.task->identifier[1];
+
```
Comment (Bidski) [inline module/behaviour/strategy/KeyboardWalk/src/KeyboardWalk.cpp:138]: Currently this line causes a set fault. Not sure why.

File: module/behaviour/strategy/KeyboardWalk/src/KeyboardWalk.cpp [hunk 2]
```diff
@@ -77,64 +120,166 @@ namespace behaviour {
...
+            });
+
+            on<Trigger<LogMessage>, Sync<KeyboardWalk>>().then([this](const LogMessage& packet) {
+                // Where this message came from
+                std::string source = "";
+
+                // If we know where this log message came from, we display that
+                if (packet.task) {
+                    // Get our reactor name
+                    std::string reactor = packet.task->identifier[1];
```
Comment (Bidski) [inline module/behaviour/strategy/KeyboardWalk/src/KeyboardWalk.cpp:137]: Currently this line causes a set fault. Not sure why.

Comment (TrentHouliston) [reply module/behaviour/strategy/KeyboardWalk/src/KeyboardWalk.cpp:137]: Not every log line has a reactor name... not sure if that would cause this though

Comment (Bidski) [reply module/behaviour/strategy/KeyboardWalk/src/KeyboardWalk.cpp:137]: It seems like the `task` pointer is being deallocated i between the if statement and `std::string reactor = packet.task->identifier[1];` 

Either that or its already an invalid pointer by the time we reach the if statement

File: module/behaviour/strategy/KeyboardWalk/src/KeyboardWalk.cpp [hunk 3]
```diff
@@ -77,64 +120,166 @@ namespace behaviour {
...
                     case 'q': quit(); return;
-                    default: log("Unknown Command");
+                    default:
+                        log<NUClear::ERROR>("Unknown Command");
+                        printStatus();
+                        break;
+                }
+            });
+
+            on<Trigger<LogMessage>, Sync<KeyboardWalk>>().then([this](const LogMessage& packet) {
```
Comment (TrentHouliston) [inline module/behaviour/strategy/KeyboardWalk/src/KeyboardWalk.cpp:130]: The Sync<KeyboardWalk> is what is causing this segfault, you can't use a reschedule operation here.

File: module/behaviour/strategy/KeyboardWalk/src/KeyboardWalk.h [hunk 4]
```diff
@@ -43,6 +46,12 @@ namespace behaviour {
             float headYaw   = 0;
             float headPitch = 0;
 
+            std::shared_ptr<WINDOW> command_window, log_window;
```
Comment (TrentHouliston) [inline module/behaviour/strategy/KeyboardWalk/src/KeyboardWalk.h:49]: Separate these out into separate lines

File: module/support/logging/FileLogHandler/data/config/docker/FileLogHandler.yaml [hunk 5]
```diff
@@ -0,0 +1 @@
+log_file: /home/nubots/NUbots/log
```
Comment (TrentHouliston) [inline module/support/logging/FileLogHandler/data/config/docker/FileLogHandler.yaml:1]: Why is this a part of this PR?

File: module/behaviour/strategy/KeyboardWalk/src/KeyboardWalk.cpp [hunk 6]
```diff
@@ -236,51 +188,81 @@ namespace behaviour {
             wrefresh(log_window.get());
         }
 
+        void KeyboardWalk::update_window(const UpdateWindow& packet) {
```
Comment (TrentHouliston) [inline module/behaviour/strategy/KeyboardWalk/src/KeyboardWalk.cpp:191]: Seems a bit excessive to have this as a struct now

File: module/support/logging/FileLogHandler/data/config/docker/FileLogHandler.yaml (hunk 5 above)
Comment (Bidski) [reply module/support/logging/FileLogHandler/data/config/docker/FileLogHandler.yaml:1]: Probably isn't very necessary, unless I also add a keyboardwalkfake role

**Comment by Bidski** [comment]
@ysims want to test this before I merge it?

**Pull request merged**